[IMP] iot: windows iot worldline exception (#7741)
odoo/documentation: content/applications/productivity/iot/config/windows_iot.rst

the :guilabel:`Rule Configuration` wizard. Now, the new rule is active and devices can connect to
the Windows virtual :abbr:`IoT (Internet of Things)` box.

Worldline exception
~~~~~~~~~~~~~~~~~~~

> Worldline is a payment terminal that can be connected to Odoo's point of sale system. It allows for
> a comprehensive and fluid payment experience for customers. Worldline is available in Benelux
> (coalition of Netherlands, Belgium, Luxembourg). When using the Windows IoT server to connect the
> Worldline payment terminal it is necessary to create an exception in the Windows firewall so that
> a connection can be made between the Odoo database/:abbr:`IoT (Internet of Things)` box and
> Worldline.

Reviewer: ```suggestion
Worldline is a payment terminal that can be connected to Odoo's point of sale system. It allows for
a comprehensive and fluid payment experience for customers. Worldline is available in Benelux
(coalition of Belgium, the Netherlands, and Luxembourg).

When using the Windows IoT server to connect the Worldline payment terminal, it is necessary to create
an exception in the Windows firewall so that a connection can be made between the Odoo
database/:abbr:`IoT (Internet of Things)` box and Worldline.
```

Reviewer: Just broke up the paragraph and reworded some stuff for clarity.

A suggestion for the first sentence, though: if the sentence "...connected to Odoo's point of sale system" is referring to the PoS app, I would word it like "...connected to Odoo's *Point of Sale* app" ... what do you think?

.. seealso::
   :doc:`../../../sales/point_of_sale/payment_methods/terminals/worldline`

To create the exception, first, open the Windows Defender Firewall app on the Windows machine. This
can be accomplished by going to the :guilabel:`Search bar` and typing in `windows defender`.

Next, click :guilabel:`Advanced settings` in the left pane.

.. image:: windows_iot/advanced-settings.png
   :align: center
   :alt: Advanced settings option highlighted in the left pane of the Windows Defender Firewall app.

> In the left pane choose, :guilabel:`Inbound Rules`.

Reviewer: ```suggestion
In the left pane, choose :guilabel:`Inbound Rules`.
```

.. image:: windows_iot/inbound-rules.png
   :align: center
   :alt: Windows Defender left window pane with inbound rules menu item highlighted.

> After selecting :guilabel:`Inbound Rules`, then select :guilabel:`New Rule` in the far right pane.

Reviewer: ```suggestion
After selecting :guilabel:`Inbound Rules`, select :guilabel:`New Rule` in the far right pane.
```

.. image:: windows_iot/new-rule.png
   :align: center
   :alt: New rule dropdown shown with new rule option highlighted.

> Next, for the :guilabel:`Rule Type` select the radio button for :guilabel:`Port`. Click
> :guilabel:`Next` to continue to the rest of the configuration.

Reviewer: ```suggestion
Then, for the :guilabel:`Rule Type`, select the radio button for :guilabel:`Port`. Click
:guilabel:`Next` to continue to the rest of the configuration.
```

.. image:: windows_iot/radio-port.png
   :align: center
   :alt: Rule Type window open, with the radio button next to port highlighted.

> On the :guilabel:`Protocols and Ports` page choose the radio button for :guilabel:`TCP` under

Reviewer: ```suggestion
On the :guilabel:`Protocols and Ports` page, choose the radio button for :guilabel:`TCP` under
```

:guilabel:`Does this rule apply to TCP or UDP?`. Next, under :guilabel:`Does this rule apply to all
local ports or specific ports?`, select the radio button for :guilabel:`Specific local ports`. Enter
`9050` and click :guilabel:`Next` to continue.

.. image:: windows_iot/protocol-port.png
   :align: center
   :alt: Protocol/port configuration window with TCP, specific port (9050) and Next highlighted.

The next screen is the :guilabel:`Action` page. Under :guilabel:`What action should be taken when a
connection matches the specified conditions?`, choose the radio button for :guilabel:`Allow the
connection`. Click :guilabel:`Next` to continue.

> A :guilabel:`Profile` page will appear. Under :guilabel:`When does this rule apply?`, leave the
> three boxes checked for: :guilabel:`Domain`, :guilabel:`Private`, and :guilabel:`Public`. Click
> :guilabel:`Next` to continue to the naming convention.

Reviewer: ```suggestion
A :guilabel:`Profile` page appears. Under :guilabel:`When does this rule apply?`, leave the
three boxes checked for: :guilabel:`Domain`, :guilabel:`Private`, and :guilabel:`Public`. Click
:guilabel:`Next` to continue to the naming convention page.
```
Present tense

On the :guilabel:`Name` page, enter `Odoo Worldline` under the :guilabel:`Name` field. Enter a

> :guilabel:`Description (optional)`. Finally click :guilabel:`Finish`.

Reviewer: ```suggestion
:guilabel:`Description (optional)`. Finally, once ready, click :guilabel:`Finish`.
```

the :guilabel:`Rule Configuration` wizard. Now, the new rule is active and devices can connect to
the Windows virtual :abbr:`IoT (Internet of Things)` box.

Worldline exception
~~~~~~~~~~~~~~~~~~~

> Worldline is a payment terminal that can be connected to Odoo's point of sale system. It allows for
> a comprehensive and fluid payment experience for customers. Worldline is available in Benelux

Reviewer 2: ```suggestion
*Worldline* is a payment terminal that can be connected to Odoo's *PoS* (point of sale) system. It
allows for a comprehensive and fluid payment experience for customers. Worldline is available in
Benelux (coalition of Belgium, the Netherlands, and Luxembourg).
```

> (coalition of Belgium, the Netherlands, and Luxembourg).

Reviewer 2: ```suggestion

```
included in the previous feedback comment

When using the Windows IoT server to connect the Worldline payment terminal, it is necessary to create

> an exception in the Windows firewall so that a connection can be made between the Odoo

Reviewer 2: ```suggestion
an exception in the Windows firewall, so that a connection can be made between the Odoo
```

database/:abbr:`IoT (Internet of Things)` box and Worldline.

.. seealso::
   :doc:`../../../sales/point_of_sale/payment_methods/terminals/worldline`

> To create the exception, first, open the Windows Defender Firewall app on the Windows machine. This
> can be accomplished by going to the :guilabel:`Search bar` and typing in `windows defender`.

Reviewer 2: ```suggestion
To create the exception, first, open the *Windows Defender Firewall* app on the Windows machine.
This can be accomplished by typing `windows defender` in the :guilabel:`Search` bar.
```

> Next, click :guilabel:`Advanced settings` in the left pane.

Reviewer 2: ```suggestion
Next, click :guilabel:`Advanced settings` in the left menu.
```

.. image:: windows_iot/advanced-settings.png
   :align: center
   :alt: Advanced settings option highlighted in the left pane of the Windows Defender Firewall app.

> In the left pane, choose :guilabel:`Inbound Rules`.

Reviewer 2: ```suggestion
In the left menu, choose :guilabel:`Inbound Rules`.
```

.. image:: windows_iot/inbound-rules.png
   :align: center
   :alt: Windows Defender left window pane with inbound rules menu item highlighted.

> After selecting :guilabel:`Inbound Rules`, select :guilabel:`New Rule` in the far right pane.

Reviewer 2: ```suggestion
After selecting :guilabel:`Inbound Rules`, select :guilabel:`New Rule` in the far right menu.
```

.. image:: windows_iot/new-rule.png
   :align: center
   :alt: New rule dropdown shown with new rule option highlighted.

Then, for the :guilabel:`Rule Type`, select the radio button for :guilabel:`Port`. Click
:guilabel:`Next` to continue to the rest of the configuration.

.. image:: windows_iot/radio-port.png
   :align: center
   :alt: Rule Type window open, with the radio button next to port highlighted.

> On the :guilabel:`Protocols and Ports` page, choose the radio button for :guilabel:`TCP` under

Reviewer 2: ```suggestion
On the :guilabel:`Protocols and Ports` page, choose the radio button for :guilabel:`TCP`, under
```

> :guilabel:`Does this rule apply to TCP or UDP?`. Next, under :guilabel:`Does this rule apply to all
> local ports or specific ports?`, select the radio button for :guilabel:`Specific local ports`. Enter
> `9050` and click :guilabel:`Next` to continue.

Reviewer 2: ```suggestion
:guilabel:`Does this rule apply to TCP or UDP?`.

Next, under :guilabel:`Does this rule apply to all local ports or specific ports?`, select the radio
button for :guilabel:`Specific local ports`. Then, enter `9050`, and click :guilabel:`Next` to
continue.
```

.. image:: windows_iot/protocol-port.png
   :align: center
   :alt: Protocol/port configuration window with TCP, specific port (9050) and Next highlighted.

The next screen is the :guilabel:`Action` page. Under :guilabel:`What action should be taken when a
connection matches the specified conditions?`, choose the radio button for :guilabel:`Allow the

> connection`. Click :guilabel:`Next` to continue.

Reviewer 2: ```suggestion
connection`. Then, click :guilabel:`Next` to continue.
```

A :guilabel:`Profile` page appears. Under :guilabel:`When does this rule apply?`, leave the
three boxes checked for: :guilabel:`Domain`, :guilabel:`Private`, and :guilabel:`Public`. Click
:guilabel:`Next` to continue to the naming convention page.

> On the :guilabel:`Name` page, enter `Odoo Worldline` under the :guilabel:`Name` field. Enter a
> :guilabel:`Description (optional)`. Finally, once ready, click :guilabel:`Finish`.

Reviewer 2: ```suggestion
On the :guilabel:`Name` page, enter `Odoo Worldline`, under the :guilabel:`Name` field. Enter a
:guilabel:`Description (optional)`. Finally, once ready, click :guilabel:`Finish`.
```

The final :guilabel:`Inbound rule` should appear as follows:

.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * - Name
     - Profile
     - Enabled
     - Action
     - Override
     - Program
     - Local Address
     - Remote Address
     - Protocol
     - Local Port
     - Remote Port
     - Authorized Users
     - Authorized Computers
     - Authorized Local Principals
     - Local User Owner
     - PolicyAppld
     - Application Package
   * - Odoo Worldline
     - All
     - Yes
     - Allow
     - No
     - Any
     - Any
     - Any
     - TCP
     - 9050
     - Any
     - Any
     - Any
     - Any
     - Any
     - None

>      - Any

Reviewer 2: you may need to speak with SALI about how to properly format this, as it is showing up a bit distorted/overlapping in my run through. Here's a screenshot of what I'm seeing...
https://drive.google.com/file/d/1_ZztD0ubjuiQf3ySLhumICpcAK--4ZG_/view?usp=sharing

Author: @samueljlieber 

When you have a moment can you take a look at this render? I would like to have the full table in this doc, if at all possible.

Author: ![2024-02-20_14-16](https://github.com/odoo/documentation/assets/91896312/8e4c1cac-70c4-4490-a5b0-273cab780af3)

I'm not showing this error when rendered and opened in Chrome.

@ksc-odoo  What browser are you viewing this in?

Reviewer 2: I use Chrome, as well. If @samueljlieber can confirm that there is no issue with the formatting, you can proceed to push this to the next stage.

Reviewer 3: Hey @tiku-odoo, when a table has too many columns it can end up overlapping/overflowing on the page because a table will only reduce its width to the contents of the cells. 

To avoid this, you can flip the axis of the table so there are many rows instead of many columns:
<img width="698" alt="Screenshot 2024-02-20 at 2 26 00 PM" src="https://github.com/odoo/documentation/assets/36018073/ed2faf91-b8cd-473a-870d-fe9b473f2c93">

```suggestion
.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * - 
     - Odoo Worldline
   * - Profile
     - All
   * - Enabled
     - Yes
   * - Action
     - Allow
   * - Override
     - No
   * - Program
     - Any
   * - Local Address
     - Any
   * - Remote Address
     - Any
   * - Protocol
     - TCP
   * - Local Port
     - 9050
   * - Remote Port
     - Any
   * - Authorized Users
     - Any
   * - Authorized Computers
     - Any
   * - Authorized Local Principals
     - Any
   * - Local User Owner
     - Any
   * - PolicyAppld
     - None
   * - Application Package
     - Any
```

the :guilabel:`Rule Configuration` wizard. Now, the new rule is active and devices can connect to
the Windows virtual :abbr:`IoT (Internet of Things)` box.

Worldline exception
~~~~~~~~~~~~~~~~~~~

*Worldline* is a payment terminal that can be connected to Odoo's *PoS* (point of sale) system. It
allows for a comprehensive and fluid payment experience for customers. Worldline is available in
Benelux (coalition of Belgium, the Netherlands, and Luxembourg).

> When using the Windows IoT server to connect the Worldline payment terminal, it is necessary to create
> an exception in the Windows firewall, so that a connection can be made between the Odoo database /
> :abbr:`IoT (Internet of Things)` box and Worldline.

Reviewer 3: Line over 100 chars
```suggestion
When using the Windows IoT server to connect the Worldline payment terminal, it is necessary to
create an exception in the Windows firewall so that a connection can be made between the Odoo
database/:abbr:`IoT (Internet of Things)` box and Worldline.
```

.. image:: windows_iot/new-rule.png
   :align: center
   :alt: New rule dropdown shown with new rule option highlighted.

Then, for the :guilabel:`Rule Type`, select the radio button for :guilabel:`Port`. Click
:guilabel:`Next` to continue to the rest of the configuration.

.. image:: windows_iot/radio-port.png
   :align: center
   :alt: Rule Type window open, with the radio button next to port highlighted.

On the :guilabel:`Protocols and Ports` page, choose the radio button for :guilabel:`TCP`, under
:guilabel:`Does this rule apply to TCP or UDP?`.

Next, under :guilabel:`Does this rule apply to all local ports or specific ports?`, select the radio
button for :guilabel:`Specific local ports`. Then, enter `9050`, and click :guilabel:`Next` to
continue.

.. image:: windows_iot/protocol-port.png
   :align: center
   :alt: Protocol/port configuration window with TCP, specific port (9050) and Next highlighted.

The next screen is the :guilabel:`Action` page. Under :guilabel:`What action should be taken when a
connection matches the specified conditions?`, choose the radio button for :guilabel:`Allow the
connection`. Then, click :guilabel:`Next` to continue.

> A :guilabel:`Profile` page appears. Under :guilabel:`When does this rule apply?`, leave the
> three boxes checked for: :guilabel:`Domain`, :guilabel:`Private`, and :guilabel:`Public`. Click
> :guilabel:`Next` to continue to the naming convention page.

Reviewer 3: Early line break
```suggestion
A :guilabel:`Profile` page appears. Under :guilabel:`When does this rule apply?`, leave the three
boxes checked for: :guilabel:`Domain`, :guilabel:`Private`, and :guilabel:`Public`. Click
```

the :guilabel:`Rule Configuration` wizard. Now, the new rule is active and devices can connect to
the Windows virtual :abbr:`IoT (Internet of Things)` box.

Worldline exception
~~~~~~~~~~~~~~~~~~~

Worldline is a payment terminal that can be connected to Odoo's point of sale system. It allows for
a comprehensive and fluid payment experience for customers. Worldline is available in Benelux
(coalition of Netherlands, Belgium, Luxembourg). When using the Windows IoT server to connect the
Worldline payment terminal it is necessary to create an exception in the Windows firewall so that
a connection can be made between the Odoo database/:abbr:`IoT (Internet of Things)` box and
Worldline.

.. seealso::
   :doc:`../../../sales/point_of_sale/payment_methods/terminals/worldline`

To create the exception, first, open the Windows Defender Firewall on the Windows machine. This can

> be accomplished by going to the :guilabel:`Search bar` and typing in `windows defender`. In the left

Reviewer 4: not that easy infortunately: first it's "Windows Defender Firewall" app
![image](https://github.com/odoo/documentation/assets/60775325/044018ee-2858-4a05-9b8b-b2e72493c7ea)
then you have to click "Advanced settings" on the left pane
![image](https://github.com/odoo/documentation/assets/60775325/80c4dbed-8c5e-415f-bd67-39f20e4314fe)
